baseapp/baseapp_test.go

```diff
@@ -139,7 +132,7 @@ func setupBaseAppWithSnapshots(t *testing.T, blocks uint, blockTxs int, options
 	snapshotStore, err := snapshots.NewStore(dbm.NewMemDB(), snapshotDir)
 	require.NoError(t, err)
 	teardown := func() {
-		os.RemoveAll(snapshotDir)
+		require.NoError(t, os.RemoveAll(snapshotDir))
```

Comment:
Why on earth do you want to make a test fail if its teardown function fails? A teardown function could fail for reasons are entirely unrelated to the test whose execution had just completed

Reply:
t.Logf() should be enough (at least it's surely better than making the test fail altogether)

Reply:
the problem with `t.Log` is that it won't be shown if the test passes. Not being able to "tidy up" after a test execution is kind of an error as well. Hence I would prefer to log it in an "always" visible way.

Reply:
What's the alternative @robert-zaremba ?

Reply:
> the problem with t.Log is that it won't be shown if the test passes.

Which is desirable I guess? Why would you want output if a test passes?

Reply:
Because it's still wrong that it can't cleanup.

Reply:
Again, that's **not** the functionality undergoing testing here! And cleanup **could** and **would quite certainly fail** for reasons that are **totally unrelated to the code undergoing testing**.

Please revert.

Reply:
BTW this could be resolved by using the following patch:

```patch
-	snapshotDir, err := ioutil.TempDir("", "baseapp")
+	snapshotDir := t.TempDir()
```

The directory would be atuomatically cleaned up when the test execution terminates.

Reply:
Really like this resolution. `t.TempDir()`++

Reply:
Yes, good idea.